pkg/ui/react-app/src/thanos/pages/blocks/SourceView.tsx

```diff
@@ -24,14 +25,29 @@ export interface SourceViewProps {
   gridMinTime: number;
   gridMaxTime: number;
   selectBlock: React.Dispatch<React.SetStateAction<Block | undefined>>;
+  blockCount: { [key: string]: number };
 }
 
-export const SourceView: FC<SourceViewProps> = ({ data, title, gridMaxTime, gridMinTime, selectBlock }) => {
+export const SourceView: FC<SourceViewProps> = ({ data, title, gridMaxTime, gridMinTime, selectBlock, blockCount }) => {
+  const linkRef: any = useRef();
+  const [ready, setReady] = useState(false);
+  const [open, setOpen] = useState(false);
+  const toggle = () => setOpen(!open);
+  useEffect(() => {
+    if (linkRef.current) {
+      setReady(true);
+    }
+  }, [linkRef.current]);
   return (
     <>
       <div className={styles.source}>
         <div className={styles.title} title={title}>
-          <span>{title}</span>
+          <span ref={linkRef}>{title}</span>
+          {ready && (
```

Comment:
```suggestion
          {linkRef.current && (
```

same as the last review, we can simply skip the `ready` state and the corresponding `useEffect` by directly using `linkRef.current` here.